### ISSUE TITLE
Accessibility: Calendar is not completely accessible with keyboard navigation

### ISSUE THREAD

**Issue by greg-a-smith** [Hacker0x01/react-datepicker#1371]
### Expected behavior
When the calendar is visible, a user should be able to access any of the controls (_e.g._ dates, month/year selection, previous/next month/year, Today button, etc.) with keyboard navigation.

### Actual behavior
Although the arrows keys do change the highlighted date, the calendar overlay itself is not accessible by keyboard navigation.

### Steps to reproduce
Go to https://reactdatepicker.com and tab into the first input so the calendar appears.

### Suggested direction
In order for the input, the calendar and the time picker to all be properly accessible with keyboard navigation, they need to be decoupled and broken into separate components.  The browser focus actually needs to move from one control to the next so proper key handling can be achieved.  The overlay also likely needs to be linked to a trigger button rather than simply tied to input focus.  When showing an overlay to pick both date and time, it needs to include some form of OK/Cancel buttons since multiple selections can (and need to) be made.

We are interested in continuing to use `react-datepicker`, but we need to make these suggested changes in order to stay compliant.  Our team is willing to help work through these issues (this one plus #1369 and #1370) with multiple pull requests, if you are open to it.

**Comment by snide** [comment]
@greg-a-smith Let me know if you get started on this. My team would like to help out as well.

**Comment by greg-a-smith** [comment]
@snide Thanks for the note.  I will be sure to reach out to you.  I was hoping to get a little more traction on the issues we logged, but we may end up just starting and then let the discussions ensue in the pull requests.

**Comment by aij** [comment]
@bcullman We don't use the time part, and I'm not sure Hacker0x01 does either...

I agree that makes things complicated though, especially since not only can you have separate date and time inputs, but you can also have a unified datetime input.
